com.unity.shadergraph/Editor/Data/Graphs/Texture2DShaderProperty.cs

```diff
@@ -40,7 +45,7 @@ internal override void ForeachHLSLProperty(Action<HLSLProperty> action)
             action(new HLSLProperty(HLSLType._Texture2D, referenceName, HLSLDeclaration.Global));
             action(new HLSLProperty(HLSLType._SamplerState, "sampler" + referenceName, HLSLDeclaration.Global));
             action(new HLSLProperty(HLSLType._float4, referenceName + "_TexelSize", decl));
-            // action(new HLSLProperty(HLSLType._float4, referenceName + "_ST", decl)); // TODO: allow users to make use of the ST values
+            action(new HLSLProperty(HLSLType._float4, referenceName + "_ST", decl)); 
```

Comment:
When scale and offset is disabled, we can skip declaring this HLSL property to save some cbuffer bandwidth

Reply:
cool, will add